mempool/src/core_mempool/transaction_store.rs

```diff
@@ -119,16 +122,28 @@ impl TransactionStore {
         }
     }
 
-    /// Fetch mempool transaction by account address + sequence_number.
-    pub(crate) fn get_mempool_txn(
-        &self,
-        address: &AccountAddress,
+    pub(crate) fn remove(
+        &mut self,
+        sender: &AccountAddress,
         sequence_number: u64,
-    ) -> Option<MempoolTransaction> {
-        self.transactions
-            .get(address)
-            .and_then(|txns| txns.get(&sequence_number))
-            .cloned()
+        is_rejected: bool,
+    ) {
+        let current_seq_number = self.sequence_numbers.get(sender).map_or(0, |v| *v);
+        if is_rejected {
+            if sequence_number >= current_seq_number {
+                self.reject_transaction(sender, sequence_number);
+            }
```

Comment:
Not an issue with this PR (cause this code existed already), but, why don't we remove the transaction if it was rejected and older than our current sequence number? Are we assuming this will never happen? (Never looked at this code, lol 😄)